[def-20] Small bugs

nomarchia opened the pull request:
[def-20-5] change language in home,orders,login - added missing translations, added multi-lang for user's "Your Orders" page. Additionally added multi-lang signs for footer;

[def-20-6] problem with language Ru /login page - refactored LoginServlet to add lang parameter if password or login was incorrect. Now russian lang works;

[def-20-8] Change "Login" to "Anmeldung" in header in german version - changed;
[def-20-9] Change "GE" to "DE" in language selection - changed;
[def-20-10] Change "Email" to "Адрес эл. почты" in russian version on login page - changed;

[def-20-11] Define font in russian version - added support for separate font if language is russian. Maryna should to choose the specific font;

[def-20-12] Synchronize languages in Admin's menu and Order page - fixed, now all signs for Admin are always in Eng;

Additional changes: used th:href instead of standard href for navigation.html (because otherwise it was working incorrectly from /orders page);

nomarchia committed:
[def-20] Small bugs:

[def-20-5] change language in home,orders,login - added missing translations, added multi-lang for user's "Your Orders" page. Additionally added multi-lang signs for footer;

[def-20-6] problem with language Ru /login page - refactored LoginServlet to add lang parameter if password or login was incorrect. Now russian lang works;

[def-20-8] Change "Login" to "Anmeldung" in header in german version - changed;
[def-20-9] Change "GE" to "DE" in language selection - changed;
[def-20-10] Change "Email" to "Адрес эл. почты" in russian version on login page - changed;

[def-20-11] Define font in russian version - added support for separate font if language is russian. Maryna should to choose the specific font;

[def-20-12] Synchronize languages in Admin's menu and Order page - fixed, now all signs for Admin are always in Eng;

Additional changes: used th:href instead of standard href for navigation.html (because otherwise it was working incorrectly from /orders page);

Rooman commented on src/main/java/com/photostudio/web/servlet/security/LoginServlet.java:
```diff
@@ -64,9 +69,9 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-                TemplateEngineFactory.process(request, response, "login", paramsMap);
+                String lang = request.getParameter("lang");
+
+                response.sendRedirect("login?lang=" + lang + "&invalid=yes");
```
сделай forward. зачем через браузер клиента гонять.

Yuliia-Sh replied:
А где lang в GET запросе используется?

nomarchia replied:
Lang не используется в GET. но раньше было, что при неправильном логине/пароле по сути снова загружалась та же страница, но уже без параметра lang, поэтому русский работал некорректно. Теперь при неправильных креденшелсах просто редиректится снова на логин, но уже с сохранением параметра языка. 

Про forward пока не понял, но погуглю

nomarchia replied:
Разобрались, откатил большинство своих изменений в LoginServlet. Оказалось, проблема была в кодировке, теперь решение переделано и фикс состоит из одной строки.

nomarchia committed:
[def-20] Small bugs:

[def-20-6] problem with language Ru /login page - refactored LoginServlet's previous code for easier and more correct solution;

Yuliia-Sh commented on src/main/webapp/WEB-INF/templates/login.html:
```diff
@@ -29,13 +29,14 @@ <h1 class="login-title">Miari Fotografie</h1>
                     </div>
                 </div>
+                <input type="hidden" name="lang" th:value="${language}">
                 <button type="submit" class="btn sign-in-button" th:text="#{sign.in.button}">Sing in</button>
```
А это теперь тоже не нужно?

nomarchia replied:
Верно. Убрано

nomarchia committed:
[def-20] Small bugs:

[def-20-6] problem with language Ru /login page - removed redundant line of code;

nomarchia committed:
Merge branch 'develop' into def-20-small-bugs-2-5-8-9-10
